ntoskrnl/mm/section.c

```diff
@@ -3641,8 +3641,43 @@ MiRosUnmapViewOfSection(IN PEPROCESS Process,
                     BaseAddress, Process, Status);
             ASSERT(NT_SUCCESS(Status));
         }
-    }
 
+        /* Flush the segment to disk, if needed */
+        PMM_SECTION_SEGMENT Segment = MemoryArea->SectionData.Segment;
+        PFILE_OBJECT FileObject = Segment->FileObject;
+
+        ASSERT(FlagOn(*Segment->Flags, MM_DATAFILE_SEGMENT));
+
+        FsRtlAcquireFileExclusive(FileObject);
+
+        /* Don't bother for auto-delete closed file. */
+        if (FlagOn(FileObject->Flags, FO_DELETE_ON_CLOSE) && FlagOn(FileObject->Flags, FO_CLEANUP_COMPLETE))
+        {
+            FsRtlReleaseFile(FileObject);
+            return STATUS_SUCCESS;
+        }
+
+        /* FIXME: We should likely flush only when last mapping is deleted */
+        PMMVAD Vad = &MemoryArea->VadNode;
+        SIZE_T ViewSize = PAGE_SIZE + ((Vad->EndingVpn - Vad->StartingVpn) << PAGE_SHIFT);
+        LARGE_INTEGER ViewOffset;
```

Comment:
Variable declarations at top of code block plz.

Reply:
Done